test/integration_tests/long/functions/test_selfdiffusion.py

```diff
@@ -0,0 +1,61 @@
+# coding=utf-8
+# Copyright 2018-2023 EvaDB
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+from test.markers import stable_diffusion_skip_marker
+from test.util import get_evadb_for_testing
+from unittest.mock import patch
+
+from evadb.server.command_handler import execute_query_fetch_all
+
+
+class StableDiffusionTest(unittest.TestCase):
+    def setUp(self) -> None:
+        self.evadb = get_evadb_for_testing()
+        self.evadb.catalog().reset()
+        create_table_query = """CREATE TABLE IF NOT EXISTS ImageGen (
+             prompt TEXT(100));
+                """
+        execute_query_fetch_all(self.evadb, create_table_query)
+
+        test_prompts = ["pink cat riding a rocket to the moon"]
+
+        for prompt in test_prompts:
+            insert_query = f"""INSERT INTO ImageGen (prompt) VALUES ('{prompt}')"""
+            execute_query_fetch_all(self.evadb, insert_query)
+
+    def tearDown(self) -> None:
+        execute_query_fetch_all(self.evadb, "DROP TABLE IF EXISTS ImageGen;")
+
+    @stable_diffusion_skip_marker
+    @patch("replicate.run", return_value=[{"response": "mocked response"}])
+    def test_stable_diffusion_image_generation(self, mock_replicate_run):
```

Comment:
Similar to test_dalle.py, this is a good unit tests.

Does StableDiffusion can run with some test api keys? If so, we can make it an integration test and use @pytest.mark.xfail(reason="") for the rate limit.

Reply:
It does run with API keys, but I figured we cannot depend on the API working every time we want to run a test. That's why I'm just checking that the API function is being at least called. I'm not familiar with  pytest.mark.xfail. If you think we can use that for the same purpose, I can check it out.

Reply:
That is to use mark flaky test. So we will run the test every time. But it will not cause the circle ci check to fail if the test failed.